.travis.yml

```diff
@@ -0,0 +1,119 @@
+# vim: set ts=2 sts=2 sw=2 expandtab :
+dist: xenial
+sudo: required
+language: bash
+services:
+  - docker
+
+before_install:
+  - curl -Ls -o docker-build https://github.com/mate-desktop/mate-dev-scripts/raw/master/travis/docker-build
+  - chmod +x docker-build
+
+install:
+  - sudo apt-get install -y python3-pip python3-setuptools
+  - sudo pip3 install --upgrade pip
+  - sudo pip install PyGithub
+  - ./docker-build --name ${DISTRO} --config .travis.yml --install
+
+script:
+  - ./docker-build --name ${DISTRO} --verbose --config .travis.yml --build autotools
+
+env:
+  - DISTRO="archlinux/base"
+  - DISTRO="debian:sid"
+  - DISTRO="fedora:29"
+  - DISTRO="ubuntu:18.10"
+
+##########################################################
+# THE FOLLOWING LINES IS USED BY docker-build
+##########################################################
+requires:
+  archlinux:
+    # Useful URL: https://git.archlinux.org/svntogit/community.git/tree/caja-extensions
+    - caja
+    - dbus-glib
+    - gcc
+    - git
+    - gupnp
+    - imagemagick
+    - intltool
+    - make
+    - mate-common
+    - python
+    - samba
+    - which
+
+  debian:
+    # Useful URL: https://github.com/mate-desktop/debian-packages
+    # Useful URL: https://salsa.debian.org/debian-mate-team/caja-extensions
+    - git
+    - libcaja-extension-dev
+    - libdbus-1-dev
+    - libdbus-glib-1-dev
+    - libdconf-dev
+    - libgail-3-dev # for caja build
+    - libgtk-3-dev
+    - libgupnp-1.0-dev
+    - libmate-desktop-dev
+    - libnotify-dev # for caja build
+    - libstartup-notification0-dev
+    - make
+    - mate-common
+    - pkg-config
+
+  fedora:
+    # Useful URL: https://src.fedoraproject.org/cgit/rpms/caja-extensions.git
+    - caja-devel
+    - dbus-glib-devel
+    - dbus-glib-devel
+    - gajim
+    - gcc
+    - git
+    - gtk3-devel
+    - gupnp-devel
+    - libSM-devel #for caja build
+    - libnotify-devel # for caja build
+    - make
+    - mate-common
+    - mate-desktop-devel
+    - redhat-rpm-config
+
+  ubuntu:
+    - git
+    - libcaja-extension-dev
+    - libdbus-1-dev
+    - libdbus-glib-1-dev
+    - libdconf-dev
+    - libgail-3-dev # for caja build
+    - libgtk-3-dev
+    - libgupnp-1.0-dev
+    - libmate-desktop-dev
+    - libnotify-dev # for caja build
+    - libstartup-notification0-dev
+    - make
+    - mate-common
+    - pkg-config
+
+variables:
+  - CFLAGS="-Wall -Werror=format-security"
+
+before_scripts:
+  - cd ${START_DIR}
+  - if [ ! -f caja-1.21.4.tar.xz ];then
```

Comment:
Not sure if we have to compile caja 1.21, but it looks like there is no problem.

Reply:
Opps, 1.20.2 might be better for 1.20 branch.